File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, arr):

        res = { }
        for i in arr:
            anagram = ''.join(sorted(i))

            if anagram in res:
                res[anagram].append(i)
            else:
                res[anagram] = [i]

        ans = []
        for a in res:
            ans.append(res[a])

        return ans

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {


    let a = {};
    let b = [];

    for( let i = 0; i < strs.length; i++){
        let c = strs[i].split('').sort().join('');

        if(c in a){
            b[a[c]].push(strs[i]);

        } else{

            a[c] = b.length;
            b.push([strs[i]]);
        }
    }
    return b;
    
};